src/mlpack/methods/decision_tree/decision_tree_impl.hpp

```diff
@@ -552,9 +596,8 @@ double DecisionTree<FitnessFunction,
       numClasses,
       UseWeights ? weights.subvec(begin, begin + count - 1) : weights);
   size_t bestDim = datasetInfo.Dimensionality(); // This means "no split".
-  DimensionSelectionType dimensions(datasetInfo.Dimensionality());
-  for (size_t i = dimensions.Begin(); i != dimensions.End();
-       i = dimensions.Next())
+  for (size_t i = dimensionSelector.Begin(); i != dimensionSelector.End();
```

Comment:
Maybe we should the same here as well: https://github.com/mlpack/ensmallen/pull/113#pullrequestreview-236418916

Reply:
Usually `End()` is just a `const size_t`, but I think paranoia is well warranted. :)  Done in 7b584bde.